ang/civicase/case/details/file-tab/directives/file-list.directive.js

```diff
@@ -4,62 +4,7 @@
   module.directive('civicaseFileList', function () {
     return {
       restrict: 'A',
-      templateUrl: '~/civicase/case/details/file-tab/directives/file-list.directive.html',
-      controller: civicaseFileListController,
-      scope: {
-        fileLists: '=civicaseFileList',
-        bulkAllowed: '='
-      }
+      templateUrl: '~/civicase/case/details/file-tab/directives/file-list.directive.html'
     };
   });
-
-  /**
-   * Controlle for File List
-   *
-   * @param {Object} $scope
-   * @param {Object} crmApi
-   * @param {Object} crmBlocker
-   * @param {Object} crmStatus
-   */
-  function civicaseFileListController ($scope, crmApi) {
-    $scope.ts = CRM.ts('civicase');
-
-    (function init () {
-      $scope.$watchCollection('fileLists.result', fileListsWatcher);
-    }());
-
-    /**
-     * Refreshes the UI state after updating the db from the api calls
-     *
-     * @params {Array} apiCalls
-     */
-    $scope.refresh = function (apiCalls) {
-      if (!_.isArray(apiCalls)) apiCalls = [];
-
-      crmApi(apiCalls, true).then(function (result) {
-        $scope.fileLists.refresh();
-      });
-    };
-
-    /**
-     * Watcher function for fileLists.result collection
-     *
-     * @params {Object} response
-     */
-    function fileListsWatcher (response) {
-      // prettier html
-      $scope.values = response.values;
-      $scope.xref = response.xref;
-      // Pre-sorting: (a) cast to array and (b) ensure stable check of isSameDate()
-      $scope.activities = response.xref ? _.sortBy(response.xref.activity, 'activity_date_time').reverse() : [];
-
-      $scope.filesByAct = {};
-      _.each(response.values, function (match) {
-        if (!$scope.filesByAct[match.activity_id]) {
-          $scope.filesByAct[match.activity_id] = [];
-        }
-        $scope.filesByAct[match.activity_id].push(response.xref.file[match.id]);
-      });
```

Comment:
why did we remove this particular function? I'm not even sure what it does.

As a side note, not sure who added this function's comment, but it doesn't explain what it does at all. I know it's a watcher, I know it's triggered when `fileLists.result`, but, what does it do?

Reply:
This was moved to `loadActivities` function https://github.com/compucorp/uk.co.compucorp.civicase/pull/480/files#diff-cf582bad70da5cf83b696953ade1a8e7R89.
This is used to fetch the list of Activities.

> As a side note, not sure who added this function's comment, but it doesn't explain what it does at all. I know it's a watcher, I know it's triggered when fileLists.result, but, what does it do?

Previously, a directive called `civicase-api3` was being used to fetch the Activities.
https://github.com/compucorp/uk.co.compucorp.civicase/pull/480/files#diff-82cd84cf830ee73a2d42893d0114f0a0L4

`civicase-api3` directive also provided sub objects like `param`, `result` or `loading`, which can be used to show loading screens and list of activities. 

But this was very confusing in my opinion, as in all other parts of the application, we dont use this. So I switched to the more usual `loadActivities` kind of function, which is much easier to understand.